releases/release-4.0.8.md

```diff
@@ -0,0 +1,121 @@
+    - Fix unexpected 'invalid auto-id' error in pessimistic txn retry. [#20134](https://github.com/pingcap/tidb/pull/20134)
+    - Fix an issue that alter enum/set type does not check constraint [#20046](https://github.com/pingcap/tidb/pull/20046)
+    - Fix cop task runtime information is wrong in the concurrent executor. [#19947](https://github.com/pingcap/tidb/pull/19947)
+    - Fix cannot select session scope explicitly with unchangeable variables [#19944](https://github.com/pingcap/tidb/pull/19944)
+
++ TiKV
+
+    - Error_code: re-format metafile [#8877](https://github.com/tikv/tikv/pull/8877)
+    - Move counter metric processing into the scheduler worker. [#8872](https://github.com/tikv/tikv/pull/8872)
+    - Fix the bug that mutex conflict of encryption makes pd-worker deal with heartbeat slow [#8869](https://github.com/tikv/tikv/pull/8869)
+    - Raftstore: record the noop entry message temporarily like pending_votes [#8864](https://github.com/tikv/tikv/pull/8864)
+    - Config: enable dynamically change config pessimistic-txn.pipelined [#8853](https://github.com/tikv/tikv/pull/8853)
+    - Add the Fast-Tune panel page as performance assisted diagnosis [#8804](https://github.com/tikv/tikv/pull/8804)
+    - Enabling profiling features by default [#8801](https://github.com/tikv/tikv/pull/8801)
+    - Fix generating memory profile [#8790](https://github.com/tikv/tikv/pull/8790)
+    - Fixed failure to backup database on GCS when storage class was provided. [#8763](https://github.com/tikv/tikv/pull/8763)
+    - Add `security.redact-info-log` config, which redacts user data from logs [#8746](https://github.com/tikv/tikv/pull/8746)
+
++ PD
+
+    - TiDB Dashboard: Fix a bug that KeyViz may panic [#3096](https://github.com/pingcap/pd/pull/3096)
+    - Generate the metafile of errors [#3090](https://github.com/pingcap/pd/pull/3090)
+    - Fix the bug that PD might panic if there is down store with 10 minutes [#3069](https://github.com/pingcap/pd/pull/3069)
+    - Add additional info for operator [#3009](https://github.com/pingcap/pd/pull/3009)
+
++ TiFlash
+
+    - Add metrics about memory usage for cop task
+    - Fix bug: waiting index during learner read may cost long time if proxy can not catch up latest raft lease info
+    - Add errors.toml to support standard error code
+    - Fix bug: proxy write too much region state info to kv engine while replaying outdated raft log
+
++ Tools
+
+    - BR
+
+        * Add errors.toml. [#563](https://github.com/pingcap/br/pull/563)
+
+    - TiCDC
+
+        * Fix a bug that processor always flushes task status to the etcd because of incorrect mod revision cache. [#1019](https://github.com/pingcap/ticdc/pull/1019)
+        * Support to replicate tables without explicit row id [#1010](https://github.com/pingcap/ticdc/pull/1010)
+        * <p><em>Sourced from <a href="https://github.com/junit-team/junit4/releases">junit's releases</a>.</em></p> [#1001](https://github.com/pingcap/ticdc/pull/1001)
```

Comment:
```suggestion
# TiCDC

## New Features

- Supports snapshot level consistency replication in time [#932](https://github.com/pingcap/ticdc/pull/932)

## Improvements

- Print statistics in MySQL sink [#1023](https://github.com/pingcap/ticdc/pull/1023)

## Bug Fixes

- Fix unexpected exit due to updating GC safepoint error [#979](https://github.com/pingcap/ticdc/pull/979)
- Fix task status is always flushed because of incorrect mod revision cache [#1017](https://github.com/pingcap/ticdc/pull/1017)
- Fix Maxwell empty messages [#978](https://github.com/pingcap/ticdc/pull/978)

# BR

## Improvements

- Speed up restore by pipeline split and restore [#428](https://github.com/pingcap/br/pull/428)
- Support restoring PD schedulers manually [#530](https://github.com/pingcap/br/pull/530)
- Use pause instead of remove schedulers [#551](https://github.com/pingcap/br/pull/551)

## Bug Fixes

- Fix send on closed channel panic during restore [#559](https://github.com/pingcap/br/pull/559)

# Dumpling

## Improvements

- Support writing directly to S3 [#155](https://github.com/pingcap/dumpling/pull/155)
- Support dumping view [#158](https://github.com/pingcap/dumpling/pull/158)
- Support dumping all generated column table [#166](https://github.com/pingcap/dumpling/pull/166)

# TiDB-Lightning

## Improvements

- Support multi bytes csv delimiter and separator [#406](https://github.com/pingcap/tidb-lightning/pull/406)
- Speed up restore by disable some pd scheduler [#408](https://github.com/pingcap/tidb-lightning/pull/408)
- Use gc ttl api for checksum gc safepoint in v4.0 cluster [#396](https://github.com/pingcap/tidb-lightning/pull/396)

## Bug Fixes

- Fix a bug about wrong column info [#420](https://github.com/pingcap/tidb-lightning/pull/420)
- Fix infinity loop in retry get region in local mode [#418](https://github.com/pingcap/tidb-lightning/pull/418)
```